src/DynamoCoreWpf/Commands/WorkspaceCommands.cs

```diff
@@ -157,5 +184,80 @@ public DelegateCommand UnPauseVisualizationManagerCommand
                 return _unpauseVisualizationManagerUpdateCommand;
             }
         }
+
+        public DelegateCommand ShowHideAllGeometryPreviewCommand
+        {
+            get
+            {
+                if (_showHideAllGeometryPreviewCommand == null)
+                {
+                    _showHideAllGeometryPreviewCommand = new DelegateCommand(
+                        ShowHideAllGeometryPreview);
+                }
+
+                return _showHideAllGeometryPreviewCommand;
+            }
+        }
+
+        public DelegateCommand ShowHideAllUpstreamPreviewCommand
+        {
+            get
+            {
+                if (_showHideAllUpstreamPreviewCommand == null)
+                {
+                    _showHideAllUpstreamPreviewCommand = new DelegateCommand(
+                        ShowHideAllUpstreamPreview);
+                }
+
+                return _showHideAllUpstreamPreviewCommand;
+            }
+        }
+
+        #endregion
+
+        #region Properties for Command Data Binding
+
+        public bool AnyNodeVisible
+        {
+            get
+            {
+                return DynamoSelection.Instance.Selection.
+                    OfType<NodeModel>().Any(n => n.IsVisible);
+            }
+        }
+
+        public bool AnyNodeUpstreamVisible
+        {
+            get
+            {
+                return DynamoSelection.Instance.Selection.
+                    OfType<NodeModel>().Any(n => n.IsUpstreamVisible);
+            }
+        }
+
+        public string ShowHideTextBubble
```

Comment:
@Benglin this property is not used in this user story but for another one.
Should we remove this?